openedx/features/specializations/views.py

```diff
@@ -38,8 +40,12 @@ def specialization_about(request, specialization_uuid):
                         key=lambda open_rerun: date_from_str(open_rerun['enrollment_start']))[0]
                         if opened else course['course_runs'][0])
         course_rerun['opened'] = opened
-        course_rerun['enrolled'] = CourseEnrollment.is_enrolled(request.user,
-                                                                CourseKey.from_string(course_rerun['key']))
+
+        course_id = CourseKey.from_string(course_rerun['key'])
+        current_course = modulestore().get_course(course_id)
+
+        course_rerun['enrolled'] = CourseEnrollment.is_enrolled(request.user, course_id)
+        course_rerun['course_first_chapter_link'] = get_course_first_chapter_link(current_course) if current_course else ""
```

Comment:
Lets check `current_course` is not `None` before extracting chapter link

Reply:
@zahid-arbisoft the check is already in place. chapter link function will get called only if `current_course` is not `None`